examples/file-uploader/js/list.js

```diff
@@ -0,0 +1,87 @@
+const Type = require('union-type');
+const T = require('ramda/src/T')
+    , adjust = require('ramda/src/adjust')
+    , append = require('ramda/src/append')
+    , curry  = require('ramda/src/curry')
+;
+const h = require('snabbdom/h');
+
+const upload = require('./upload');
+const uploader = require('./uploader');
+
+const sync = (s) => [s, []];
```

Comment:
`noEffect` is more accurate. You could have synchronous effects for whatever reason. I think `noFx` is the name Elm suggests using.

The bigger issue I see is parent components have to know when to expect effects from any descendant components. If only one action has a side-effect, then the signature of *all* update return values has to be changed accordingly, and on up the tree. Maybe it's ok, but somehow it seems implicit and untyped. 

(I noticed in your timesheet app you have split `update` and `updateE`, but have not studied it enough to understand if this is an alternative way of dealing with this issue?)